langchain/client/runner_utils.py

```diff
@@ -50,14 +54,26 @@ class InputFormatError(Exception):
     """Raised when the input format is invalid."""
 
 
-def _get_prompts(inputs: Dict[str, Any]) -> List[str]:
-    """Get prompts from inputs.
+## Shared Utilities
+
+
+def _first_example(examples: Iterator[Example]) -> Tuple[Example, Iterator[Example]]:
+    """Get the first eample while chaining it back and preserving the iterator."""
+    try:
+        example: Example = next(examples)
+    except StopIteration:
+        raise ValueError("No examples provided.")
+    return example, itertools.chain([example], examples)
+
+
+def _get_prompt(inputs: Dict[str, Any]) -> str:
```

Comment:
Previously was using the generate() API and permitted batch runs which doesn't make sense when we want to evaluate. Change slightly so that we evaluate only single prompts/single list of messages and use the predict / predict_messages api